File: proyect/proyect/src/main/java/taller2backend/proyect/service/ProveedorService.java
package taller2backend.proyect.service;
import java.util.List;
import java.util.Optional;

import taller2backend.proyect.dto.*;

public interface ProveedorService {


    //listar proveedores
    List<ProveedorDTO> listarProveedores();

    //crear proveedor
    public ProveedorDTO crearProveedor(ProveedorDTO proveedorDto);
    
    //actualizar proveedor
    public ProveedorDTO actualizarProveedor(Long  id, ProveedorDTO proveedorDto);
    //eliminar proveedor
    public void eliminarProveedor(Long id);

    //listar proveedores por nombre
    public List<ProveedorDTO> listarProveedoresPorNombre(String nombre);

    //obtener proveedor por id
    Optional<ProveedorDTO> obtenerPorId(Long id);
}

File: proyproyecto/src/service/ProveedorService.js
import axios from 'axios'   

const API_URL = 'http://localhost:9090/api/proveedor'

const apiProveedor = axios.create({
    baseURL: API_URL,
    headers: {
        'Content-Type': 'application/json'
    }
})


export default{
    
    //listar proveedores
    listProveedor(){
        return apiProveedor.get()
    },

    //bucar provedores por nombre
    buscarPorNombre(nombre){
        return apiProveedor.get(`/buscar?nombre=${nombre}`)
    },
    
    //obtener proveedor por id
    obtenerPorId(id){
        return apiProveedor.get(`/${id}`)
    },
    
    //crear proveedor
    crearProveedor(proveedor){
        return apiProveedor.post('/crear',proveedor)
    },
    
    //actualizar proveedor
    actualizarProveedor(id,proveedor){
        return apiProveedor.put(`/${id}`,proveedor)
    },

    //eliminar proveedor
    eliminarProveedor(id){
        return apiProveedor.delete(`/${id}`)
    }
}